core/transientstore/store.go

```diff
@@ -80,11 +86,45 @@ type RwsetScanner struct {
 
 // NewStoreProvider instantiates TransientStoreProvider
 func NewStoreProvider(path string) (StoreProvider, error) {
-	dbProvider, err := leveldbhelper.NewProvider(&leveldbhelper.Conf{DBPath: path})
+	// Ensure the routine is invoked while the peer is down.
+	lockPath := filepath.Join(filepath.Dir(path), "fileLock")
+	lock := leveldbhelper.NewFileLock(lockPath)
+	if err := lock.Lock(); err != nil {
+		return nil, errors.WithMessage(err, "as another peer node command is executing,"+
+			" wait for that command to complete its execution or terminate it before retrying")
+	}
+
+	provider, err := newStoreProvider(path, lock)
 	if err != nil {
-		return nil, err
+		return nil, errors.WithMessagef(err, "could not construct storage provider in folder [%s]", path)
```

Comment:
For code sanity you could add `lock.Unlock()` before returning error -- as at this level of code it's not fair to assume that the consumer may not retry to reopen the store.

Reply:
got it.